pkg/sync/compare.go

```diff
@@ -81,18 +82,19 @@ func CompareGeneratedAssets(rootFs billy.Filesystem, newCharts, newAssets, origi
 			for _, f := range fileInfos {
 				chartVersion := f.Name()
 				splitChartVersion := strings.Split(chartVersion, "-rc")
-				chartVersionWithoutRC := strings.Join(splitChartVersion[:len(splitChartVersion)-1], "")
-				latestRCSeenSoFar, ok := latestRC[chartVersionWithoutRC]
+				chartVersionWithName := fmt.Sprintf("%s/%s", chart, chartVersion)
+				chartVersionWithNameWithoutRC := fmt.Sprintf("%s/%s", chart, splitChartVersion[0])
```

Comment:
Are you confident they will all have RC's ? And that this [0] doesn't hit out-of-bounds.

Reply:
Even if they don't have an RC, splitChartVersion[0] will just contain the packageVersion of the chart. However, this isn't an edge case I've fully accounted for; I'm not sure how not having RCs will impact https://github.com/rancher/charts-build-scripts/blob/5d6fca48e3d5366200999599c9d60f24a3318ddd/pkg/sync/compare.go#L92-L94.

What should the expected behavior be when RC exist but a packageVersion without the RC also exists on the staging branch?

Reply:
If there is a new package without RC I'd assume that is the new production version ? As long as it doesn't also exist in release branch which would be invalid.

Reply:
Yeah, packages without RC shouldn't exist in the Staging branch in the first place, since it belongs in production. Expanding the scope of the Staging branch to also have packageVersions not scoped to a specific rcVersion isn't something we're tracking right now.

You also would never be able to push something to the Staging Branch that has a version in the Live Branch since the Source Branch validates against both Staging and Live (so it would fail on validating against Live since the packageVersion is already tracked).

Reply:
Does that mean then for pulling in an upstream non-rancher package at 3.1.0, you have to tag it as 3.1.0-rc1 in staging ? That works but its kinda a misnomer.

Reply:
Yes, except it would be 3.1.000-rc1 (where 00 is the packageVersion)